src/gui/log/loglistview.cpp

```diff
@@ -0,0 +1,124 @@
+            const int textHMargin = style->pixelMetric(QStyle::PM_FocusFrameHMargin, &option, option.widget) + 1;
+            const int textVMargin = style->pixelMetric(QStyle::PM_FocusFrameVMargin, &option, option.widget);
+            const QRect textRect = style->subElementRect(QStyle::SE_ItemViewItemText, &option, option.widget)
+                                   .adjusted(textHMargin, textVMargin, -textHMargin, -textVMargin);
+
+            QPen coloredPen = originalPen;
+            coloredPen.setColor(Qt::gray);
+            painter->setPen(coloredPen);
+            const QString time = index.data(BaseLogModel::TimeRole).toString();
+            style->drawItemText(painter, textRect, 0, option.palette, (option.state & QStyle::State_Enabled), time);
+
+            painter->setPen(originalPen);
+            const int timeWidth = option.fontMetrics.width(time);
+            style->drawItemText(painter, textRect.adjusted(timeWidth, 0, 0, 0), 0, option.palette, (option.state & QStyle::State_Enabled), SEPERATOR);
+
+            const QVariant foreground = index.data(Qt::ForegroundRole);
+            if (foreground.canConvert<QColor>()) { // peermodel may give invalid foreground role
+                coloredPen.setColor(foreground.value<QColor>());
+                painter->setPen(coloredPen);
+            }
+
+            style->drawItemText(painter, textRect.adjusted((timeWidth + option.fontMetrics.width(SEPERATOR)), 0, 0, 0), 0
+                                , option.palette, (option.state & QStyle::State_Enabled), index.data(BaseLogModel::MessageRole).toString());
+            painter->restore();
+        }
+
+        QSize sizeHint(const QStyleOptionViewItem &option, const QModelIndex &index) const override
+        {
+            const QString displayText = QString::fromLatin1("%1%2%3").arg(index.data(BaseLogModel::TimeRole).toString(), SEPERATOR
+                                                                          , index.data(BaseLogModel::MessageRole).toString());
+            return QSize(option.fontMetrics.width(displayText), QStyledItemDelegate::sizeHint(option, index).height());
+        }
+    };
+}
+
+LogListView::LogListView(QWidget *parent)
+    : QListView(parent)
+{
+    setSelectionMode(QAbstractItemView::ExtendedSelection);
+    setItemDelegate(new LogItemDelegate(this));
+
+#if defined(Q_OS_MAC)
+    setAttribute(Qt::WA_MacShowFocusRect, false);
+#endif
+}
+
+void LogListView::keyPressEvent(QKeyEvent *event)
+{
+    if (event->matches(QKeySequence::Copy))
+        copySelection();
+    else
+        QListView::keyPressEvent(event);
+}
+
+void LogListView::copySelection() const
+{
+    QStringList list;
+    const QModelIndexList selectedIndexes = selectionModel()->selectedRows();
+    for (const QModelIndex &index : selectedIndexes) {
+        list.append(QString::fromLatin1("%1%2%3").arg(index.data(BaseLogModel::TimeRole).toString(), SEPERATOR
```

Comment:
I would have separator inside format string to avoid yet another argument insertion. @Chocobo1?